scenario.py

```diff
@@ -16,26 +15,49 @@
 import operator
 import random
 
+from dataset_and_model import DatasetAndModel
 import constants
 from partner import Partner
-import utils
 
 
 class Scenario:
     def __init__(self, params, experiment_path, scenario_id=1, n_repeat=1):
 
-        # Identify and get a dataset for running experiments
-        self.dataset_name = "MNIST"
-        (x_train, y_train), (x_test, y_test) = mnist.load_data()
-        self.nb_samples_used = len(x_train)
+        # Get and verify which dataset is configured
+        supported_datasets = ["mnist", "cifar10"]
```

Comment:
`supported_dataset_names` to be consistent. 
Or `dataset `instead of `dataset_name`

Reply:
Ok